### PR TITLE
tikv-server: allow set timeout and hearbeat ticks

### PR DESCRIPTION
**Pull request by BusyJay**
So we can change ticks without recompilation.

@siddontang @hhkbp2 PTAL

### CODE REVIEW

File: src/bin/tikv-server.rs [hunk 1]
```diff
@@ -448,6 +452,15 @@ fn build_cfg(matches: &Matches, config: &toml::Value, cluster_id: u64, addr: Str
...
+    cfg_usize(&mut cfg.raft_store.raft_heartbeat_ticks,
+              config,
+              "raftstore.raft-heartbeat-ticks");
+    if cfg_usize(&mut cfg.raft_store.raft_election_timeout_ticks,
+                 config,
+                 "raftstore.raft-election-timeout-ticks") {
+        warn!("Election timeout ticks needs to be same across all the cluster, otherwise it may \
+               lead to inconsistency.");
+    }
     cfg_u64(&mut cfg.raft_store.split_region_check_tick_interval,
```
Comment (siddontang) [inline src/bin/tikv-server.rs]: we should also check election ticks > heartbeat ticks.

File: src/raftstore/store/config.rs [hunk 2]
```diff
@@ -152,6 +152,14 @@ impl Config {
     }
 
     pub fn validate(&self) -> Result<()> {
+        if self.raft_heartbeat_ticks == 0 {
+            return Err(box_err!("heartbeat tick must greater than 0"));
+        }
+
+        if self.raft_election_timeout_ticks <= self.raft_heartbeat_ticks {
```
Comment (siddontang) [inline src/raftstore/store/config.rs]: do we allow equal here? 
Election should be more several times bigger than heartbeat, maybe we can define a max multiple like 5.

Comment (BusyJay) [inline src/raftstore/store/config.rs]: This is just a copy-paste of https://github.com/pingcap/tikv/blob/master/src/raft/raft.rs#L138

**Comment by hhkbp2** [comment]
LGTM

**Comment by BusyJay** [comment]
@siddontang PTAL

File: etc/config-template.toml [hunk 3]
```diff
@@ -73,6 +73,12 @@ split-region-check-tick-interval = "5s"
 # It's recommanded to set it to 3/4 of region-split-size.
 # raft-log-gc-size-limit = "48MB"
 
+# Send a heartbeat message after every configured number of ticks.
```
Comment (siddontang) [inline etc/config-template.toml]: remove this from config file, we don't want user to try it.

**Comment by siddontang** [comment]
Rest LGTM

**Pull request merged**